docs/source/api_reference/database.rst

```diff
@@ -0,0 +1,144 @@
+
+For example, if a ``Service`` has both a record in a MySQL database, and a
+``job_registration_schema`` stored on a MongoDB instance, the ``Service``
+model class would be responsible for ensuring that the MongoDB record and
+the MySQL record agree.
+
+Models
+------
+
+The models in this package use SQLAlchemy's object-relational mapping (ORM)
+capabilities to acquire a persistent representation in a relational database
+. The API makes use of these model classes to obtain a representation of a
+resource in the database, as well as to describe the relations between
+models. For example, a ``Service`` has a one-to-many relationship between
+``Job``s, expressed by a ``FOREIGN KEY`` constraint in the relational database.
+
+Abstract Database Model
+~~~~~~~~~~~~~~~~~~~~~~~
+
+.. automodule:: topchef.database.models.abstract_database_model
+    :members:
+    :private-members:
+    :special-members:
+    :exclude-members: __dict__, __weakref__, __module__
+
+Service
+~~~~~~~
+
+.. automodule::  topchef.database.models.service
+    :members:
+    :private-members:
+    :special-members:
+    :exclude-members: __dict__, __weakref__, __module__
+
+Job
+~~~
+
+.. automodule:: topchef.database.models.job
+    :members:
+    :private-members:
+    :special-members:
+    :exclude-members: __dict__, __weakref__, __module__
+
+Job Set
+~~~~~~~
+
+.. automodule:: topchef.database.models.job_set
+    :members:
+    :private-members:
+    :special-members:
+    :exclude-members: __dict__, __weakref__, __module__
+
+
+Schemas
+-------
+
+The schema of a database is the set of all metadata that enables records to
+be consistently entered into that database. By "consistent", what we mean is
+ that any operation not resulting in an error maps the database from one
+ valid state to another valid state. Unlike some no-SQL databases like
+ MongoDB, relational databases require that a schema is defined before
+ writing any records to the database. This module takes care of defining
+ that schema, using abstractions provided by SQLAlchemy. Based on this
+ schema and the particular database being used, SQLAlchemy can then generate
+  the SQL code needed to perform a particular manipulation on a database model.
+
+Abstract Database Schema
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. automodule:: topchef.database.schemas.abstract_database_schema
+    :members:
+    :private-members:
+    :special-members:
+    :exclude-members: __dict__, __weakref__, __module__
+
+Database Schema
+~~~~~~~~~~~~~~~
+
+.. automodule:: topchef.database.schemas.database_schema
+    :members:
+    :private-members:
+    :special-members:
+    :exclude-members: __dict__, __weakref__, __module__
+
+Job Status
+~~~~~~~~~~
+
+.. automodule:: topchef.database.schemas.job_status
+    :members:
+    :private-members:
+    :special-members:
+    :exclude-members: __dict__, __weakref__, __module__
+
+JSON Type
+---------
+
+SQL-based databases are strongly and statically typed. This means that every
```

Comment:
Check that these paragraphs aren't in the ``json_type`` module documentation. It seems like that would be a much better place for it.

Also, AFAIK Travis-CI doesn't have MySQL 5.7, where they introduced the JSON data type. Could you check whether the JSON gets stored as a JSON type in a MySQL database?